code/modules/reagents/reagent_containers/blood_pack.dm

```diff
@@ -67,6 +67,21 @@
 	else
 		..()
 
+/obj/item/weapon/reagent_containers/blood/attackby(obj/item/weapon/P as obj, mob/user as mob)
+	..()
+	if (istype(P, /obj/item/weapon/pen))
+		if (reagents.get_reagent_amount("blood") && name != "Empty BloodPack") //Stops people mucking with bloodpacks that are filled
+			usr << "<span class='notice'>You can't relabel [name] until it is empty!</span>"
+			return
+		var/blood_name = input(usr, "What blood type would you like to label it as?", "Blood Types") in list("A+", "A-", "B+", "B-", "O+", "O-", "AB+", "AB-", "Cancel")
+		if (blood_name == "Cancel") return
+		var/obj/item/i = usr.get_active_hand()
+		if (!istype(i, /obj/item/weapon/pen) || !in_range(user, src)) return //Checks to see if pen is still held or bloodback is in range
+		name = "BloodPack [blood_name]"
+		desc = "Contains blood used for transfusion."

```

Comment:
If this necessary? Or is the initial value of desc for the blood pack something else?